client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -1561,11 +1561,11 @@ static XContentType enforceSameContentType(IndexRequest indexRequest, @Nullable
     /**
      * Utility class to build request's endpoint given its parts as strings
      */
-    static class EndpointBuilder {
+    public static class EndpointBuilder {
```

Comment:
maybe if Params and EndpontBuilder are also used externally, they should be extracted as top-level classes? WDYT?

Reply:
I think they are really only useful in the context of other request converters so I'm fine with them staying inner classes.